internal/outpost/ldap/search.go

```diff
@@ -79,3 +80,52 @@ func (ls *LDAPServer) fallbackRootDSE(req *search.Request) (ldap.ServerSearchRes
 		Referrals: []string{}, Controls: []ldap.Control{}, ResultCode: ldap.LDAPResultSuccess,
 	}, nil
 }
+
+func (ls *LDAPServer) Compare(boundDN string, req ldap.CompareRequest, conn net.Conn) (ldap.LDAPResultCode, error) {
+	log.WithField("CompareRequest", req).Info()
+
+	// search for entry referred to by req.DN
+	searchReq := ldap.NewSearchRequest(req.DN, ldap.ScopeBaseObject, ldap.DerefAlways, 0, 0, true, "(objectClass=*)", []string{}, []ldap.Control{})
+	searchResult, err := ls.Search(boundDN, *searchReq, conn)
+	if err != nil {
+		return searchResult.ResultCode, err
+	}
+	if len(searchResult.Entries) != 1 {
+		return ldap.LDAPResultNoSuchObject, nil
+	}
+
+	// transform assertions into a filter string
+	// TODO: Create the ber.Packet directly, instead of relying so heavily on ldap.CompileFilter
+	safe := regexp.MustCompile(`[\w\s]*`)
```

Comment:
```suggestion
	safe := regexp.MustCompile(`^[\w\s]*$`)
```
Are these necessary here?

EDIT 2026.08: Yes, they are. adding them in next push.